spec/presenters/tree_builder_tags_spec.rb

```diff
@@ -0,0 +1,98 @@
+describe TreeBuilderTags do
+  before do
+    role = MiqUserRole.find_by_name("EvmRole-operator")
+    @group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Tags Group")
+    login_as FactoryGirl.create(:user, :userid => 'tags_wilma', :miq_groups => [@group])
+    @tag_selected = FactoryGirl.create(:classification, :name => 'tag_selected', :show => false)
+    @folder_selected = FactoryGirl.create(:classification, :name => 'folder_selected', :show => true)
+    @folder_selected.entries.push(@tag_selected)
+    @tag_not_selected = FactoryGirl.create(:classification, :name => 'tag_not_selected', :show => false)
+    @folder_not_selected = FactoryGirl.create(:classification, :name => 'folder_not_selected', :show => true)
+    @folder_not_selected.entries.push(@tag_not_selected)
+    @filters = {"#{@folder_selected.name}-#{@tag_selected.name}" =>
+                "/managed/#{@folder_selected.name}/#{@tag_selected.name}"}
+    @group = FactoryGirl.create(:miq_group)
+  end
+  context 'read-only mode' do
+    before do
+      edit = nil
+      @tags_tree = TreeBuilderTags.new(:tag,
+                                       :tag_tree,
+                                       {},
+                                       true,
+                                       :edit => edit, :filters => @filters, :group => @group)
+    end
+    it 'set init options correctly' do
+      tree_options = @tags_tree.send(:tree_init_options, :tags)
+      expect(tree_options).to eq(:full_ids => true, :add_root => false, :lazy => false)
+    end
+    it 'set locals for render correctly' do
+      locals = @tags_tree.send(:set_locals_for_render)
+      expect(locals[:id_prefix]).to eq('tags_')
+      expect(locals[:checkboxes]).to eq(true)
+      expect(locals[:check_url]).to eq("ops/rbac_group_field_changed/#{@group.id || "new"}___")
```

Comment:
@martinpovolny It is changed. Please have a look at https://github.com/ManageIQ/manageiq/pull/10904/files#diff-10b0e2e7cca6cd95ed278244856045f1R33